rts/Game/Camera.cpp

```diff
@@ -795,8 +795,16 @@ float3 CCamera::NearTheaterIntersection(const float3& dir, const float rayLength
 	if (pos.y < maxAltitude)
 		return pos;
 
-	const auto fv1 = TracePointToMaxAltitude(GetFrustumVert(CCamera::FRUSTUM_POINT_FBL), rayLength, maxAltitude);
-	const auto fv2 = TracePointToMaxAltitude(GetFrustumVert(CCamera::FRUSTUM_POINT_FBR), rayLength, maxAltitude);
+	const auto fbl = GetFrustumVert(CCamera::FRUSTUM_POINT_FBL);
+	const auto fbr = GetFrustumVert(CCamera::FRUSTUM_POINT_FBR);
+	const auto ftl = GetFrustumVert(CCamera::FRUSTUM_POINT_FTL);
+
+	// check the bottom frustum is parallel to ground and lower than the top frustum
+	if ((std::abs(fbl.y-fbr.y) > std::numeric_limits<float>::epsilon()) || (fbl.y >= ftl.y))
```

Comment:
`epscmp` exists, perhaps use that? Maybe `std::numerlic_limits<T>::epsilon()` could even become the default for its 3rd arg:
https://github.com/beyond-all-reason/spring/blob/a2dd68209e8cb636b3c5a90b07cc929e45f03598/rts/System/SpringMath.h#L141